integration/ide/bsp-server/src/BspServerTests.scala

```diff
@@ -0,0 +1,164 @@
+package mill.integration
+
+import ch.epfl.scala.{bsp4j => b}
+import mill.api.BuildInfo
+import mill.bsp.Constants
+import mill.testkit.BspServerUtil._
+import mill.testkit.UtestIntegrationTestSuite
+import utest._
+
+import scala.jdk.CollectionConverters._
+
+object BspServerTests extends UtestIntegrationTestSuite {
+  def fixturePath: os.Path =
+    super.workspaceSourcePath / "fixtures"
+  override protected def workspaceSourcePath: os.Path =
+    super.workspaceSourcePath / "project"
+
+  def tests: Tests = Tests {
+    test("BSP install") - integrationTest { tester =>
+      import tester._
+      eval("mill.bsp.BSP/install", stdout = os.Inherit, stderr = os.Inherit, check = true)
+
+      withBspServer(
+        workspacePath,
+        millTestSuiteEnv
+      ) { (buildServer, initRes) =>
+        val scalaVersion = sys.props.getOrElse("MILL_TEST_SCALA_2_13_VERSION", ???)
+        val scalaTransitiveSubstitutions = {
+          val scalaFetchRes = coursierapi.Fetch.create()
+            .addDependencies(coursierapi.Dependency.of(
+              "org.scala-lang",
+              "scala-compiler",
+              scalaVersion
+            ))
+            .fetchResult()
+          scalaFetchRes.getDependencies.asScala
+            .filter(dep => dep.getModule.getOrganization != "org.scala-lang")
+            .map { dep =>
+              def basePath(version: String) =
+                s"${dep.getModule.getOrganization.split('.').mkString("/")}/${dep.getModule.getName}/$version/${dep.getModule.getName}-$version"
```

Comment:
This interpolated string is way too long and complicated, let's break it up into some local variables that we combined into a string at the end